examples/octet_stream/text-client.cpp

```diff
@@ -0,0 +1,140 @@
+
+namespace {
+
+std::atomic<bool> shutdown_flag;
+
+void set_shutdown_flag(int) {
+  shutdown_flag = true;
+}
+
+} // namespace
+
+int caf_main(caf::actor_system& sys, const config& cfg) {
+  namespace ssl = caf::net::ssl;
+  // Do a regular shutdown for CTRL+C and SIGTERM.
+  auto default_term = signal(SIGTERM, set_shutdown_flag);
+  auto default_sint = signal(SIGINT, set_shutdown_flag);
+  // Read the configuration.
+  auto use_ssl = caf::get_or(cfg, "tls.enable", false);
+  auto port = caf::get_or(cfg, "port", default_port);
+  auto host = caf::get_or(cfg, "host", default_host);
+  auto ca_file = caf::get_as<std::string>(cfg, "tls.ca-file");
+  caf::actor producer_actor;
+  caf::actor consumer_actor;
+  auto conn
+    = caf::net::octet_stream::with(sys)
+        // Optionally enable TLS.
+        .context(ssl::context::enable(use_ssl)
+                   .and_then(ssl::emplace_client(ssl::tls::v1_2))
+                   .and_then(ssl::load_verify_file_if(ca_file)))
+        // Connect to "$host:$port".
+        .connect(host, port)
+        // If we don't succeed at first, try up to 10 times with 1s delay.
+        .retry_delay(1s)
+        .max_retry_count(9)
+        // After connecting, spin up a worker that prints received inputs.
+        .start([&sys, &producer_actor, &consumer_actor](auto pull, auto push) {
+          consumer_actor = sys.spawn([pull](caf::event_based_actor* self) {
+            pull.observe_on(self)
+              .transform(caf::flow::byte::split_as_utf8_at('\n'))
+              .do_on_error([self](const caf::error& err) {
+                self->println("*** connection error: {}", err);
+              })
+              .do_finally([self] {
+                self->println("*** lost connection to server -> quit\n*** use "
+                              "CTRL+D or CTRL+C to terminate");
+                self->quit();
+              })
+              .for_each([self](const caf::cow_string& line) {
+                self->println("reply: {}", line.str());
+              });
+          });
+          producer_actor
+            = sys.spawn([push](caf::event_based_actor*) -> caf::behavior {
+                auto producer = caf::async::make_blocking_producer(push);
+                assert(producer);
+                return {
+                  [producer = std::move(producer)](
+                    const std::string& msg) mutable {
+                    producer->push(caf::as_bytes(caf::make_span(msg)));
+                  },
+                };
+              });
+        });
+  // Report any error to the user.
+  if (!conn) {
+    sys.println("*** unable to run at port {}: {}", port, conn.error());
+    return EXIT_FAILURE;
+  }
+  // Wait for CTRL+C or SIGTERM.
+  sys.println("*** Client is running, press CTRL+C to stop");
+  auto line = std::string{};
+  while (std::getline(std::cin, line)) {
+    line = line + '\n';
+    caf::anon_mail(line).send(producer_actor);
```

Comment:
This breaks backpressure. Use a blocking producer here. Then, we'll only need to spawn one worker actor that does both flows. Also means we no longer need to care about the actor handles, since the blocking producer will close the flow for the worker at scope exit.